packages/components/src/components/trace/TraceFilter.vue

```diff
@@ -0,0 +1,354 @@
+        <ArrowSearchRightIcon />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import ArrowSearchLeftIcon from '@/assets/arrow-search-left.svg';
+import ArrowSearchRightIcon from '@/assets/arrow-search-right.svg';
+import CloseThinIcon from '@/assets/close-thin.svg';
+import SearchIcon from '@/assets/search.svg';
+import { getEventTarget } from '@appland/diagrams';
+
+export default {
+  name: 'v-trace-filter',
+
+  components: {
+    ArrowSearchLeftIcon,
+    ArrowSearchRightIcon,
+    CloseThinIcon,
+    SearchIcon,
+  },
+
+  props: {
+    nodesLength: {
+      type: Number,
+      require: false,
+      default: 0,
+    },
+    currentIndex: {
+      type: Number,
+      default: null,
+    },
+    suggestions: {
+      type: Array,
+      default: () => [],
+    },
+  },
+
+  data() {
+    return {
+      filterValue: '',
+      showSuggestions: false,
+      selectedSuggestion: 0,
+    };
+  },
+
+  watch: {
+    filterValue: {
+      handler(value) {
+        this.$emit('onChange', value);
+      },
+    },
+    selectedSuggestion: {
+      handler(selectedSuggestion) {
+        const selected =
+          this.$refs.suggestionsList.querySelectorAll('li')[selectedSuggestion];
+
+        if (selected.offsetTop < this.$refs.suggestionsList.scrollTop) {
+          this.$refs.suggestionsList.scrollTop = selected.offsetTop;
+        } else if (
+          selected.offsetTop >=
+          this.$refs.suggestionsList.scrollTop +
+            this.$refs.suggestionsList.offsetHeight
+        ) {
+          this.$refs.suggestionsList.scrollTop =
+            selected.offsetTop - selected.offsetHeight * 2;
+        }
+      },
+    },
+  },
+
+  computed: {
+    formClasses() {
+      const classList = ['trace-filter__form'];
+
+      if (this.showSuggestions && this.suggestions.length) {
+        classList.push('trace-filter__form--show-suggestions');
+      }
+
+      return classList;
+    },
+  },
+
+  methods: {
+    setValue(newValue) {
+      this.filterValue = newValue;
+    },
+    clearValue() {
+      this.filterValue = '';
+    },
+    onPrevArrow() {
+      this.$emit('onPrevArrow');
+    },
+    onNextArrow() {
+      this.$emit('onNextArrow');
+    },
+    getSuggestionClasses(index) {
+      const classes = ['trace-filter__suggestions-item'];
+      if (index === this.selectedSuggestion) {
+        classes.push('trace-filter__suggestions-item--selected');
+      }
+      return classes;
+    },
+    onInputFocus() {
+      this.showSuggestions = true;
+      this.selectedSuggestion = 0;
+    },
+    onFormSubmit() {
+      if (
+        this.$refs.input === window.document.activeElement &&
+        this.showSuggestions &&
+        this.suggestions.length
+      ) {
+        this.makeSelection(this.suggestions[this.selectedSuggestion].id);
+      }
+    },
+    makeSelection(eventId) {
+      this.filterValue = `${this.filterValue} id:${eventId}`.trim();
+      this.showSuggestions = false;
```

Comment:
The text input is still focused after submitting, so the user is free to type another query, but they'll have no suggestions for the second query

https://user-images.githubusercontent.com/8737782/152193174-0147524c-3d2d-4900-8802-7b0c1dfbaa7b.mp4

.

Reply:
Fixed https://github.com/applandinc/appmap-js/pull/534/commits/16d5e5d02a8ac984eb15a64001f3398079c10f74